blmath/geometry/convexify.py

```diff
@@ -12,19 +18,49 @@ def convexify_planar_curve(polyline, flatten=False, want_vertices=False):
         - want_vertices:
             Boolean; do you want the indices to the convex hull vertices?
     '''
-    import numpy as np
-    from scipy.spatial import ConvexHull  # First thought this warning was caused by a pythonpath problem, but it seems more likely that the warning is caused by scipy import hackery. pylint: disable=no-name-in-module
-    from blmath.geometry import Polyline
-    from blmath.geometry.transform.rotation import rotate_to_xz_plane
-    from blmath.geometry.transform.translation import translation
+    if flatten:
+        return convexify_and_flatten_planar_curve(polyline, want_vertices, normal)
+
+    v = polyline.v
+    if len(v) <= 1:
+        return polyline
+
+    if normal is None:
+        normal = estimate_normal(v)
+
+    # ConvexHull(v) requires that v be projected to a plane and made N x 2.
+    # This is done by simply dropping whichever dimension is closest to the normal.
+    dim_to_drop = np.argmax(np.abs(normal))
```

Comment:
@algrs If you drop a dimension, it's true, you are projecting on to a different plane, but I realized that if you take a convex hull you'll get the same set of points even if you project on to the "right" plane